# Predicting ICB Therapy Response Using Integrative Data Sources
## Project Overview
This project aims to develop a machine learning framework that predicts the response to Immune Checkpoint Blockade (ICB) therapy by integrating various genomic, transcriptomic, and clinical data sources. The model utilizes data such as gene expression, SNP mutation, and clinical data to enhance the prediction accuracy.

## Features
- Integration of multiple genomic data sources.
- Implementation of feature selection techniques.
- Use of machine learning models like Random Forest, Support Vector Machines, and Logistic Regression.
- Evaluation metrics including accuracy, F1 score, and ROC AUC score.
## Getting Started
### Prerequisites
Python 3.x
Pandas
Numpy
Scikit-learn
Matplotlib
Scipy
### Installation
Clone the repository to your local machine:
git clone https://github.com/ahmedhesham47/Machine-Learning-Integrative-Framework-for-Predicting-ICB-Response/
### Usage
Navigate to the project directory and run the Jupyter notebook:
cd path/to/your-repository
jupyter notebook
## Data
The project uses three main types of data:
Gene Expression Data
SNP Mutation Data
CNA Mutation
Clinical Data
The datasets (Liu and Ravi) are provided in this repository in the Data Folder.

## Models
The project explores several machine-learning models:

RandomForestClassifier
LogisticRegression
Support Vector Machine (SVM)

## Contributing
Contributions to this project are welcome. Please fork the repository and submit a pull request.
